src/org/openstreetmap/josm/gui/tagging/presets/TaggingPresetSelector.java

```diff
@@ -264,14 +268,12 @@ public void actionPerformed(ActionEvent ae) {
     protected synchronized void filterItems() {
         //TODO Save favorites to file
         String text = edSearchText.getText().toLowerCase(Locale.ENGLISH);
-        boolean onlyApplicable = ckOnlyApplicable != null && ckOnlyApplicable.isSelected();
-        boolean inTags = ckSearchInTags != null && ckSearchInTags.isSelected();
-        boolean isDeprecated = ckDeprecated != null && ckDeprecated.isSelected();
 
         DataSet ds = OsmDataManager.getInstance().getEditDataSet();
         Collection<OsmPrimitive> selected = (ds == null) ? Collections.<OsmPrimitive>emptyList() : ds.getSelected();
         final List<PresetClassification> result = classifications.getMatchingPresets(
-                text, onlyApplicable, inTags, isDeprecated, getTypesInSelection(), selected);
+                text, ONLY_APPLICABLE.getPref(PresetSearchFilter.ONLY_APPLICABLE), SEARCH_IN_TAGS.getPref(PresetSearchFilter.SEARCH_IN_TAGS),
+                DEPRECATED_TAGS.getPref(PresetSearchFilter.DEPRECATED_TAGS), getTypesInSelection(), selected);
```

Comment:
Ideally, you would just use the same EnumMap from one of my other comments (`EnumMap<PresetSearchFilter, JCheckBox>`) instead.

Example: `map.entrySet().stream().filter(entry -> entry.getValue().isSelected().map(Map.Entry::getKey).toArray(PresetSearchFilter[]::new)`.